mesonbuild/compilers.py

```diff
@@ -2729,3 +2733,135 @@ def unix_compile_flags_to_native(self, args):
 
     def get_link_debugfile_args(self, targetfile):
         return []
+
+
+# Only tested on linux for ICC 17.0.0
+#    * object extraction does not compile (don't know) (25 object extraction fails)
```

Comment:
I can look into this, but we need a simple way to install ICC. We will also need to add it to the CI.